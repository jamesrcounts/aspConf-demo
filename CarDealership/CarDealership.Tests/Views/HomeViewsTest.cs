using System;
using System.Web.Mvc;
using ApprovalTests.Asp.Mvc;
using ApprovalTests.Reporters;
using CarDealership.Controllers;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CarDealership.Tests.Views
{
    [TestClass]
    [UseReporter(typeof(TortoiseDiffReporter), typeof(FileLauncherReporter))]
    public class HomeViewsTest
    {
        [TestMethod]
        public void IndexViewTest()
        {
            // make the view available

            // provide route
            ApprovalTests.Asp.PortFactory.MvcPort = 50978;
            var controller = new HomeController();
            Func<ActionResult> method = controller.Index;

            // test it (red)
            MvcApprovals.VerifyMvcPage(method);

            // review it approve if appropriate
            // test it (green)
            // refactor.
        }

        [TestMethod]
        public void AboutViewTest()
        {
            // aboutViewTest
        }

        [TestMethod]
        public void CarsViewTest()
        {
            // some test with data, should be ineresting
        }
    }
}