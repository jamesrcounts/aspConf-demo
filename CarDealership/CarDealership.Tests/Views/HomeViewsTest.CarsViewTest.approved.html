<!DOCTYPE html>
<html>
<head><base href="http://localhost:42111/Home">
    <meta charset="utf-8" />
    <title>Cars</title>
    <link href="/Content/Site.css" rel="stylesheet" type="text/css" />
    <script src="/Scripts/jquery-1.5.1.min.js" type="text/javascript"></script>
    <script src="/Scripts/modernizr-1.7.min.js" type="text/javascript"></script>
</head>
<body>
    <div class="page">
        <header>
            <div id="title">
                <h1>My MVC Application</h1>
            </div>

            <nav>
                <ul id="menu">
                    <li><a href="/">Home</a></li>
                    <li><a href="/Home/About">About</a></li>
                </ul>
            </nav>
        </header>
        <section id="main">
            

<h2>Cars</h2>

<p>
    <a href="/Home/Create">Create New</a>
</p>
<table>
    <tr>
        <th>
            Make
        </th>
        <th>
            Model
        </th>
        <th>
            Year
        </th>
        <th></th>
    </tr>

    <tr>
        <td>
            Vw
        </td>
        <td>
            Passat
        </td>
        <td>
            2002
        </td>
        <td>
            <a href="/Home/Edit">Edit</a> |
            <a href="/Home/Details">Details</a> |
            <a href="/Home/Delete">Delete</a>
        </td>
    </tr>
    <tr>
        <td>
            Honda
        </td>
        <td>
            Fit
        </td>
        <td>
            2007
        </td>
        <td>
            <a href="/Home/Edit">Edit</a> |
            <a href="/Home/Details">Details</a> |
            <a href="/Home/Delete">Delete</a>
        </td>
    </tr>

</table>

        </section>
        <footer>
        </footer>
    </div>
</body>
</html>
